app/models/account/provider_methods.rb

```diff
@@ -154,7 +154,7 @@ def buyer_alerts
 
   def provider_key
     ensure_provider
-    bought_cinstances.first!.user_key
+    bought_cinstances.reset.first!.user_key
```

Comment:
My feeling is that there was a problem somewhere, where stale `bought_cinstances` were creating troubles. I think it is more appropriate to call `provider.reset_bought_cinstances` at that place instead of making a new database call every time `provider.provider_key` is called.

Reply:
I think I did that at first, but then found more and more cases where this would be needed, and decided to fix it at source, basically to guarantee that it will not only work in tests, but also in real requests.

Reply:
I'm not sure I agree here. Failing here means that something is changing the `bought_cinstances` underneath the current process which I don't expect in production and where it is expected, should be protected. Because it is not only `provider_key` that can be affected by such unexpected modifications.

It's introducing an additional SQL query and any other attribute may suffer. e.g. why `provider_key` would require resetting but not lets say `first_service!`?

If there are many test failures, I can help investigating and fixing them.